### ISSUE TITLE
Lantern loses light but keeps burning after falling in water

### ISSUE THREAD

**Issue by mugg91** [transhumandesign/kag-base#1217]
**Description**

A Lantern can stop emitting light but still keep its burning animation after touching water.
This happens if the following is true:
- Playing on online server
- Lantern was either spawned in by !lantern, or it was already there when the map loaded.

It does not happen:
- In local modes, regardless if spawned in or already there.
- In online servers, if buying the Lantern from a builder shop.

**Reproduction**

One way to do it:
- Join online Sandbox server such as "Official Sandbox EU".
- Type in !lantern.
- Throw it in water.
- Notice that it doesn't emit light in dark places but still has the burning animation sprite. 

Or another way to do it:
- Join online Challenges server such as "Official Challenges US".
- Play until you get the map that has lanterns and water.
- Hold it and enter the water.
- Notice that it keeps burning.

**Screenshots**

https://i.imgur.com/SY4Fbqi.png

**Additional context**

Posting this issue at Nananas' request.

**Comment by eps0003** [comment]
Was about to comment that your steps to reproduce on sandbox don't work, but after rejoining the server the lanterns revert back to their burning animation.

**Comment by mugg91** [comment]
It looks like my understanding of the bug was not quite correct.

While testing again, I found that the bug only happens some of the time. In other words, some lanterns I dropped in water were extinguished properly while others weren't.

On top of that, there is the chance that the bug will not happen at all. So all lanterns dropped in water will be extinguished properly. In this case, rejoining the server may incur the bug happening again.

It doesn't matter if you buy from buildershop or spawn it in. Not sure about lanterns that were already loaded at start.
In offline solo Sandbox the bug still doesn't happen.

Maybe it has to do with the server not catching up, lagging behind, etc.
